tests/Validation/ValidationTest.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace LanguageServer\Tests;
+
+use Exception;
+use LanguageServer\Definition;
+use LanguageServer\Index\Index;
+use LanguageServer\ParserKind;
+use LanguageServer\PhpDocument;
+use LanguageServer\DefinitionResolver;
+use phpDocumentor\Reflection\DocBlock;
+use phpDocumentor\Reflection\DocBlockFactory;
+use PHPUnit\Framework\TestCase;
+use LanguageServer\ClientHandler;
+use LanguageServer\Protocol\Message;
+use AdvancedJsonRpc;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use Sabre\Event\Loop;
+use Microsoft\PhpParser;
+
+$frameworksDir = realpath(__DIR__ . '/../../validation/frameworks');
+
+class ValidationTest extends TestCase
```

Comment:
Added a comment, made the test name better, and cleaned it up some more.

I think this is an easy and powerful way to write tests for some cases, but not every case. With a little work, it could be extended to test more scenarios with this strategy.